### DISCUSSION TITLE
As any user I would like to get better search results in homepage user search.

### DISCUSSION THREAD

**Discussion by Gummibeer** [Astrotomic/opendor.me#31] [Ideas] [answered]
As _any user_ I would like to get better search results in the homepage user search.

<img width="325" alt="Bildschirmfoto_2021-04-11_um_23 19 22" src="https://user-images.githubusercontent.com/6187884/114321787-e0edc300-9b1c-11eb-9cc4-a22d42c1d93f.png">

**Comment by Gummibeer** [comment]
https://github.com/Astrotomic/opendor.me/commit/c854f69de2a6ec62bfae3b76c3e3e6141b2b66a7

**Reply by aidan-casey** [reply]
It seems like a lot of optimization could be done on the server and as you continue to add more users, loading all of them into a single JSON response seems like it could slow down the results. How do you feel about offsetting this to API?

For example:

```php
Route::get('user/autocomplete', static function (Request $request) {
    $term = $request->input('search') . '%';

    $result = User::query()
        ->select('id', 'name', 'display_name')
        ->whereIsRegistered()
        ->where(function($query) use ($term) {
            $query->where('name', 'like', $term);
            $query->orWhere('display_name', 'like', $term);
        })
        ->limit(3)
        ->get();

    return response()->json($result);
})->name('user.autocomplete');
```

**Reply by Gummibeer** [reply]
This whole thing will be offloaded to the generous people at https://algolio.com 

So Laravel Scout, Algolia, instantsearch.js

This will also power the coming developer, organization, repo search.

**Reply by aidan-casey** [reply]
Makes sense. Feel free to ping me if you want assistance. 👍🏼

**Comment by Gummibeer** [comment, marked as answer]
https://github.com/Astrotomic/opendor.me/pull/85
